### PR TITLE
Guardian brushup and bullet passthrough

### PR DESCRIPTION
**Pull request by CrudeWax**
## About the PR 
Fixes the unintentional design issue of holoparasites not being immune to bullets. 
Now bullets pass through holoparasites and other forms of guardians, with the intention being that security will have to either mix up their arsenal with lasers, or shoot the owner, hence introducing more counter-play than "Rush with lethal gun"

Code-wise adds magic ~~and power miner (renamed to salvage warden)~~ guardian varieties, as well as re-writes the guardians to have a base instead of it just being holoparasite.
image

**Screenshots**

**Changelog**

:cl: CrudeWax
- fix: Bullets now fly through Guardians in accordance with the Cybersun handbook. Better aim for the owner!
- tweak: Stun bolts also pass through guardians. Laser beams however don't.

### CODE REVIEW

**Comment by CrudeWax** [comment]
Why are pipes haunting my PRs

**Comment by moonheart08** [comment]
We are not giving miners/salvagers holoparasites.

**Comment by mirrorcult** [comment]
just gut it from the PR for now

**Comment by CrudeWax** [comment]
Removed

**Comment by mirrorcult** [comment]
fix conflicts and good to merge

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Most of the sprite fields are dupe with the base one.

**Comment by CrudeWax** [comment]
Ready for merge

File: Resources/Prototypes/Entities/Objects/Devices/Syndicate_Gadgets/guardian_activators.yml [hunk 1]
```diff
@@ -9,7 +9,23 @@
...
     netsync: false
   - type: GuardianCreator
-    guardianProto: MobHoloparasite
+    guardianProto: MobHoloparasiteGuardian
+
+- type: entity
+  name: magical lamp
+  id: MagicalLamp
+  parent: BaseItem
+  description: I guess the wizard federation had to cut costs after the jhin merchandise scandal
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Devices/Syndicate_Gadgets/guardian_activators.yml:18]: League reference?

File: Resources/Prototypes/Entities/Mobs/Player/guardian.yml [hunk 2]
```diff
@@ -42,9 +43,7 @@
         radius: 0.35
       mass: 10
       mask:
-      - Impassable
-      - SmallImpassable
-      - MobImpassable
+      - VaultImpassable
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Mobs/Player/guardian.yml:46]: Are these correct?

File: Resources/Prototypes/Entities/Objects/Devices/Syndicate_Gadgets/guardian_activators.yml [hunk 3]
```diff
@@ -9,7 +9,23 @@
...
+  components:
+  - type: Sprite
+    sprite: Objects/Misc/Lights/lamp.rsi
+    layers:
+      - state: lamp
+      - state: lamp-on
+        shader: unshaded
+        visible: false
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Devices/Syndicate_Gadgets/guardian_activators.yml:26]: ?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Bunch of duplicate datafields and also spelling.

File: Resources/Prototypes/Entities/Mobs/Player/guardian.yml (hunk 2 above)
Comment (CrudeWax) [reply Resources/Prototypes/Entities/Mobs/Player/guardian.yml:46]: Yes.

File: Resources/Prototypes/Entities/Objects/Devices/Syndicate_Gadgets/guardian_activators.yml (hunk 3 above)
Comment (CrudeWax) [reply Resources/Prototypes/Entities/Objects/Devices/Syndicate_Gadgets/guardian_activators.yml:26]: As intended.

**Comment by Mith-randalf** [comment]
Out of curiosity, can you run me through the reason for holoparas being immune to bullets?

**Comment by mirrorcult** [comment]
> Out of curiosity, can you run me through the reason for holoparas being immune to bullets?

lets them fill a more interesting niche, gives them a bit more utility since they're underpowered now, and makes combat with holopara users more fun

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**